embeddings/mlx_embeddings.py

```diff
@@ -0,0 +1,205 @@
+# Based on the MLX BERT implementation from mlx-examples, but
+# designed to easily pull any HuggingFace BERT embeddings model
+# and use it for batch inference for text embeddings.
+# It also implements the proper API to be evaluated in MTEB
+# and used like a sentence transformer model.
+import torch
+from typing import Literal
+import mlx.core as mx
+import mlx.nn as nn
+from mlx.utils import tree_unflatten, tree_flatten, tree_map
+from dataclasses import dataclass
+from typing import Optional
+import numpy as np
+from transformers import BertTokenizer, BertConfig, AutoModel
+
+def replace_key(key: str) -> str:
+    key = key.replace(".layer.", ".layers.")
+    key = key.replace(".self.key.", ".key_proj.")
+    key = key.replace(".self.query.", ".query_proj.")
+    key = key.replace(".self.value.", ".value_proj.")
+    key = key.replace(".attention.output.dense.", ".attention.out_proj.")
+    key = key.replace(".attention.output.LayerNorm.", ".ln1.")
+    key = key.replace(".output.LayerNorm.", ".ln2.")
+    key = key.replace(".intermediate.dense.", ".linear1.")
+    key = key.replace(".output.dense.", ".linear2.")
+    key = key.replace(".LayerNorm.", ".norm.")
+    key = key.replace("pooler.dense.", "pooler.")
+    return key
```

Comment:
Could you rename the modules in the individual layers to avoid this key mapping? (I know some other examples e.g. BERT do this, but I've been trying to move everything to use the same structure as Transformers as it really simplifies things.